### ISSUE TITLE
100ms audio delay

### ISSUE THREAD

**Issue by djrooli** [rigaya/NVEnc#164]
Hello Rigaya!

I'm struggling with conversion again :)

I noticed some videos I tried to encode has audio delay of 100 ms.
After conversion if I play the video file and add 100 ms + of audio delay It syncs.
So probably the first 100 ms of video frames are missing?

I haven't checked it yet in a video editing software.
Do you have any experience with this delay?
I checked the finished video in SPEK but I can't see difference in the audio.

It happened before, so it is not your new version of NVEncC.

THX!

**Comment by rigaya** [comment]
I've checked some videos, but the number of input frames and number of output frames are the same, so there should be no video frames missing, as far as the input video starts from IDR frame.

- AudioVideo Sync has been improved in NVEnc 4.30, so please use NVEnc 4.30 or later.
- When the input video is in VFR (Variable Frame Rate), you might need to try "--avsync vfr".
- If the frame rate is not detected correctly, you might need to try settings frame rate explicitly, by "--fps".

If the list above does not work, it's rather difficult problem. Generally, these AVSync issues depends on the input file, it'll be difficult to check without the actual input file.

**Comment by djrooli** [comment]
Thank you for your efforts in regarding the issue.

I am using your cl you have sent me, but I had to switch to cfr in some cases to maintain quality.
I always use avsync depends on what I'm using. VFR or CFR. In this case it was force cfr.
Larger videos are always good, so yes, it depends on the source. The source this time was a DVD series muxed to MKV with MakeMKV. Anyway, I added a 100ms + delay in mkvtoolnix and it is fine now.

And I forgot to tell you, as I mentioned I got this error from before so I managed somehow to archieve non delayed audio changing the cl, but I didn't save it. So next time I will have issue like this I upload the video file too.

I close this issue for now and open a new if I can provide full details.

Thank you again!
